### PR TITLE
fix: request.stream edge cases …

### PR DESCRIPTION
**Pull request by ronag**
Fixes some edge case TODOs

### CODE REVIEW

File: test/client-stream.js [hunk 1]
```diff
@@ -184,29 +184,3 @@ test('stream GET remote destroy', (t) => {
...
-    t.expectUncaughtException()
-    client.stream({
-      path: '/',
-      method: 'GET'
-    }, (err, { statusCode, headers }) => {
-      t.error(err)
-      return new PassThrough()
-    })
-  })
-})
```
Comment (ronag) [inline test/client-stream.js:212]: This test should have been removed with the API change.

**Comment by ronag** [comment]
Node v10 failure... Node 10 is end of life. Could we remove it?

**Comment by mcollina** [comment]
This probably needs tests

File: lib/client.js [hunk 2]
```diff
@@ -62,21 +62,9 @@ class Parser extends HTTPParser {
...
-        if (EventEmitter.errorMonitor) {
-          body.once(EventEmitter.errorMonitor, this.resume)
-        } else {
-          body.once('error', (err) => {
-            this.resume()
-            if (body.listenerCount('error') === 0) {
-              body.emit('error', err)
-            }
-          })
-        }
```
Comment (ronag) [inline lib/client.js:79]: The `error` stuff no longer do anything so there is nothing to test.

File: lib/request.js [hunk 3]
```diff
@@ -138,20 +138,17 @@ function dispatch (err, data) {
...
-        const callback = this.callback
-        this.callback = null
-        callback(null, null)
       }
     } catch (err) {
-      if (!this.callback) {
-        throw err
-      } else {
-        this.callback(err, null)
-      }
```
Comment (ronag) [inline lib/request.js:153]: The `this.callback` stuff just moves things around a bit, nothing to test.

File: lib/client.js [hunk 4]
```diff
@@ -62,21 +62,9 @@ class Parser extends HTTPParser {
...
+          .once('error', this.resume)
```
Comment (ronag) [inline lib/client.js:67]: We will always either emit `'error'` unless gracefully ended. That's the only case we need resume.

File: lib/request.js [hunk 5]
```diff
@@ -138,20 +138,17 @@ function dispatch (err, data) {
     try {
       const stream = this.factory(data)
       if (stream) {
-        finished(stream, this.callback)
+        finished(stream, (err) => {
+          stream.destroy(err)
```
Comment (ronag) [inline lib/request.js:142]: This is a bit of an edge case I'm unsure how to trigger exactly in a test... will need to give it more thought.

**Comment by ronag** [comment]
Found some other issues while trying to figure out some tests. Wip

**Comment by ronag** [comment]
@mcollina PTAL, added test and found another issue which is resolved
